### PR TITLE
Make sure root item is created before lambda gets called for desktop …

### PR DESCRIPTION
**Pull request by vladest**
…loader
https://highfidelity.fogbugz.com/f/cases/7518/Build-7104-Breaks-Settings-Menus

Test plan:

- load interface. open menu in Tablet or Desktop modes
- make sure menu is visible and functional

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3194/HighFidelity-Beta-PR11335-dfaa8761e5d723403220b8b23c0608df126a4029.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3194/HighFidelity-Beta-PR11335-dfaa8761e5d723403220b8b23c0608df126a4029.exe

**Comment by Nex-Pro** [comment]
Testing this

File: libraries/ui/src/ui/OffscreenQmlSurface.cpp [hunk 1]
```diff
@@ -704,7 +703,20 @@ void OffscreenQmlSurface::finishQmlLoad(QQmlComponent* qmlComponent, QQmlContext
...
+    if (newItem) {
+        // Make sure we make items focusable (critical for
+        // supporting keyboard shortcuts)
+        newItem->setFlag(QQuickItem::ItemIsFocusScope, true);
+    }
+
+    // Make sure we will call callback for this codepath
+    // Caal this before qmlComponent->completeCreate() otherwise ghost window appears
+    if (newItem && _rootItem) {
+        f(qmlContext, newObject);
```
Comment (samcake) [inline libraries/ui/src/ui/OffscreenQmlSurface.cpp]: single letter function/lambda ? not good...

Comment (sethalves) [reply libraries/ui/src/ui/OffscreenQmlSurface.cpp]: yeah maybe we should fix that and fix the comment Caal --> Call
This is blocking some people, so I didn't want to be too picky.

Comment (Nex-Pro) [reply libraries/ui/src/ui/OffscreenQmlSurface.cpp]: I agree

Comment (AndrewMeadows) [reply libraries/ui/src/ui/OffscreenQmlSurface.cpp]: In this case he is just moving the single-letter argument. It isn't new, and that pattern is used in many places in this file.  Perhaps fixing that variable name should be done more thoroughly in a separate PR.

**Comment by Nex-Pro** [comment]
Testing with win10 and vive.
HMD and desktop.

**Comment by Nex-Pro** [comment]
Menu is now displaying in HMD and desktop in tablet and toolbar modes.

In the course of checking I noticed that there are some issues remaining with the menus.
I will check fogbugz for each issue.

Got a crash each time I selected 'Display' from the menu ( was in HMD Tablet) 
Each time I restarted interface it was not in HMD. 
Also crashes this way from HMD Toolbar when 'Display' is selected from menu.
No splats generated.
Logs attached.

Does not crash if 'Display' is selected from the menu in HMD.

Using 'Exit VR' button does not trigger crash.

Unable to check this against dev latest due to the menu issue.

### LOGS

[Logs.zip](https://github.com/highfidelity/hifi/files/1293462/Logs.zip)

**Comment by Nex-Pro** [comment]
### Changes in the build above should have no effect on results posted earlier.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3198/HighFidelity-Beta-PR11335-34c1cb6579b88d3099411d8123745f6ad113263c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3198/HighFidelity-Beta-PR11335-34c1cb6579b88d3099411d8123745f6ad113263c.exe

**Comment by vladest** [comment]
@Nex-Pro is crash is reproducible in non HMD mode?

**Comment by anshuman64** [comment]
@Nex-Pro @vladest crash occurs in Build 7103 (the last build where menus are working).

Therefore, this PR does not cause it.

Calling this a QA pass and merging.

**Comment by anshuman64** [comment]
Ticket crash issue: https://highfidelity.fogbugz.com/f/cases/7539/CRASH-In-tablet-HMD-go-to-Menu-Settings-Display

**Pull request merged**